src/org/usfirst/frc2130/visionTargetting/Robot.java

```diff
@@ -0,0 +1,263 @@
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.Scheduler;
+import edu.wpi.first.wpilibj.livewindow.LiveWindow;
+import edu.wpi.first.wpilibj.networktables.NetworkTable;
+
+import edu.wpi.first.wpilibj.vision.VisionRunner;
+import edu.wpi.first.wpilibj.vision.VisionThread;
+
+import java.awt.geom.Arc2D.Float;
+import java.io.File;  
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.channels.Pipe;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.HashMap;
+
+import edu.wpi.first.wpilibj.vision.VisionPipeline;
+import edu.wpi.first.wpilibj.vision.VisionThread;
+import gripvision.GripPipeline;
+
+import org.opencv.core.*;
+import org.opencv.core.Core.*;
+import org.opencv.features2d.FeatureDetector;
+import org.opencv.imgcodecs.Imgcodecs;
+import org.opencv.imgproc.*;
+import org.opencv.objdetect.*;
+
+import org.opencv.core.Mat;
+import org.opencv.imgproc.Imgproc;
+import org.usfirst.frc2130.visionTargetting.commands.*;
+import org.usfirst.frc2130.visionTargetting.subsystems.*;
+
+import edu.wpi.cscore.CvSink;
+import edu.wpi.cscore.CvSource;
+import edu.wpi.cscore.UsbCamera;
+import edu.wpi.first.wpilibj.CameraServer;
+import edu.wpi.first.wpilibj.IterativeRobot;
+
+import java.io.File; 
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.HashMap;
+
+import org.opencv.core.*;
+import org.opencv.core.Core.*;
+import org.opencv.features2d.FeatureDetector;
+import org.opencv.imgcodecs.Imgcodecs;
+import org.opencv.imgproc.*;
+import org.opencv.objdetect.*;
+
+
+/**
+ * The VM is configured to automatically run this class, and to call the
+ * functions corresponding to each mode, as described in the IterativeRobot
+ * documentation. If you change the name of this class or the package after
+ * creating this project, you must also update the manifest file in the resource
+ * directory.
+ */
+
+
+
+public class Robot extends IterativeRobot {
+	
+
+
+	NetworkTable table;
+
+    Command autonomousCommand;
+    
+	public VisionThread visionThread;
+	
+	public static  Object imgLock;
+	
+	public static double centerX;
+	
+	public static double run;
+	
+
+	private static final int IMG_WIDTH = 320;
+	private static final int IMG_HEIGHT = 240;
+
+	private boolean run1;
+    
+    public static OI oi;
+    // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+    public static robotDrive robotDrive;
+
+    // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+
+    /**
+     * This function is run when the robot is first started up and should be
+     * used for any initialization code.
+     */
+    @Override
+    public void robotInit() {
+    RobotMap.init();  
+    	
+    // Creates UsbCamera and MjpegServer [1] and connects them
+
```

Comment:
The block of code from here down is from a different video capture example.

This step calls a static method on the CameraServer object to get a single global instance of the camera instead of instantiating a new one each time.

See:

http://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/CameraServer.html#startAutomaticCapture

The return from this function is a UsbCamera instance. I have some more comments below about the function. Note that you can assign the output to UsbCamera variable:

`UsbCamera camera = CameraServer.getInstance().startAutomaticCapture();`